test/e2e/antreapolicy_test.go

```diff
@@ -2410,6 +2410,86 @@ func testFQDNPolicy(t *testing.T) {
 	time.Sleep(networkPolicyDelay)
 }
 
+func testFQDNPolicyInClusterService(t *testing.T) {
+
+	var services []*v1.Service
+	if clusterInfo.podV4NetworkCIDR != "" {
+		ipv4Svc := k8sUtils.BuildService("ipv4-svc", "x", 80, 80, map[string]string{"pod": "a"}, nil)
+		ipv4Svc.Spec.ClusterIP = "None"
+		ipv4Svc.Spec.IPFamilies = []v1.IPFamily{v1.IPv4Protocol}
+		services = append(services, ipv4Svc)
+	}
+	if clusterInfo.podV6NetworkCIDR != "" {
+		ipv6Svc := k8sUtils.BuildService("ipv6-svc", "x", 80, 80, map[string]string{"pod": "b"}, nil)
+		ipv6Svc.Spec.ClusterIP = "None"
+		ipv6Svc.Spec.IPFamilies = []v1.IPFamily{v1.IPv6Protocol}
+		services = append(services, ipv6Svc)
+	}
+
+	for _, service := range services {
+		k8sUtils.CreateOrUpdateService(service)
+		failOnError(waitForResourceReady(service, timeout), t)
+	}
+
+	svcDNSName := func(service *v1.Service) string {
+		return fmt.Sprintf("%s.%s.svc.cluster.local", service.Name, service.Namespace)
```

Comment:
nit: you don't need `.cluster.local`, which is cluster dependent

Reply:
FQDN will use exact-match on this case. I guess to reuse this func we can use `%s.%s.svc.cluster.local` in both policy and probe?

Reply:
sorry I don't understand: what's the issue with using `<name>.<namespace>.svc` everywhere?

Reply:
DNS response uses the full name `<name>.<namespace>.svc.cluster.local`. If FQDN use `<name>.<namespace>.svc`, it doesn't think this DNS response matched.

Reply:
Good observation, thanks. It means I need to update the example in my PR: https://github.com/antrea-io/antrea/pull/2667 :)
I wonder if we should do something about this in the future. Probably not very important since FQDN should mostly be used for external traffic.